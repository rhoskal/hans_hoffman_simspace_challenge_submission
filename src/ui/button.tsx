import React from "react";
import styled from "styled-components";

type Props = {
  isActive?: boolean;
  onClick?: () => void;
  text?: string;
};

const StyledButton = styled.button<{ isActive: Boolean }>`
  background-color: ${props => (props.isActive ? null : "#d4d4d4")};
  background-image: ${props =>
    props.isActive ? "linear-gradient(to right, #348AC7 51%, #7474BF 100%)" : null};
  color: ${props => (props.isActive ? "white" : "black")};
  border: none;
  padding: 10px;
  width: 100%;
  border-radius: 4px;
  text-align: center;
  text-decoration: none;
`;

export function Button({ isActive = false, onClick, text = "Click Me" }: Props) {
  return (
    <StyledButton disabled={isActive} isActive={isActive} onClick={onClick}>
      {text}
    </StyledButton>
  );
}
