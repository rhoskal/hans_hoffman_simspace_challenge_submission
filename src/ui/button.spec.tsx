import React from "react";
import { fireEvent, render } from "@testing-library/react";

import { Button } from "./button";

describe("[Component]: Button", function() {
  let onClick: () => void;
  beforeEach(function() {
    onClick = jest.fn();
  });

  it("should disabled onClick", function() {
    const text: string = "pitbull";
    const { getByText } = render(<Button isActive onClick={onClick} text={text} />);

    const btn = getByText(text);

    fireEvent.click(btn);

    expect(onClick).not.toHaveBeenCalled();

    // https://github.com/styled-components/jest-styled-components/issues/298
    // expect(btn).toHaveStyleRule("color", "white");
  });

  it("should handle onClick", function() {
    const text: string = "pitbull";
    const { getByText } = render(<Button onClick={onClick} text={text} />);

    fireEvent.click(getByText(text));

    expect(onClick).toHaveBeenCalled();
  });

  it("should match snapshot", function() {
    const { container } = render(<Button />);

    expect(container).toMatchSnapshot();
  });
});
