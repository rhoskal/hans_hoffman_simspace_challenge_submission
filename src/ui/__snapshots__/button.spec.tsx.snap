// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component]: Button should match snapshot 1`] = `
<div>
  <button
    class="sc-AxjAm itLOjr"
  >
    Click Me
  </button>
</div>
`;
